### PR TITLE
Inline a couple of 2D grid cell access functions

### PR DESCRIPTION
**Pull request by ojura**
Noticed `ToFlatIndex` during profiling. Inline a couple of other short functions as well.

### CODE REVIEW

File: cartographer/mapping/2d/grid_2d.h [hunk 1]
```diff
@@ -70,7 +78,10 @@ class Grid2D : public GridInterface {
   // Grows the map as necessary to include 'point'. This changes the meaning of
   // these coordinates going forward. This method must be called immediately
   // after 'FinishUpdate', before any calls to 'ApplyLookupTable'.
-  virtual void GrowLimits(const Eigen::Vector2f& point);
+  virtual void GrowLimits(const Eigen::Vector2f& point) {
```
Comment (ojura) [inline cartographer/mapping/2d/grid_2d.h]: Just noticed this is a virtual method. Perhaps inlining it does not make any sense.

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 2.

**Comment by MichaelGrupp** [comment]
FYI: I unblocked "Verify pull request"

**Comment by wally-the-cartographer** [comment]
Error: exit status 1

Log:
~~~
=> handle work item: PR 1420 on googlecartographer/cartographer.
 => Switching to branch master.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at 36ea74d9 Stop using broken async_grpc::Client retry strategy
 $ git clean -fd [data/cartographer]
 $ git checkout master [data/cartographer]
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
 $ git pull [data/cartographer]
From https://github.com/googlecartographer/cartographer
   e1a839d6..d0348634  master     -> origin/master
Updating e1a839d6..d0348634
Fast-forward
 cartographer/mapping/2d/grid_2d.cc | 21 ---------------------
 cartographer/mapping/2d/grid_2d.h  | 18 +++++++++++++++---
 cmake/functions.cmake              |  4 ++++
 3 files changed, 19 insertions(+), 24 deletions(-)
 => Checking out ojura/patch-1 as local branch pr_1420.
 $ git remote add ojura git@github.com:ojura/cartographer.git [data/cartographer]
fatal: remote ojura already exists.
 $ git fetch ojura [data/cartographer]
 $ git branch --track pr_1420 ojura/patch-1 [data/cartographer]
Branch 'pr_1420' set up to track remote branch 'patch-1' from 'ojura'.
 => Switching to branch pr_1420.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at d0348634 Enable -Wthread-safety for Clang and CMake. (#1403)
 $ git clean -fd [data/cartographer]
 $ git checkout pr_1420 [data/cartographer]
Switched to branch 'pr_1420'
Your branch is up to date with 'ojura/patch-1'.
 $ git pull --force [data/cartographer]
Your configuration specifies to merge with the ref 'refs/heads/patch-1'
from the remote, but no such ref was fetched.

~~~

**Pull request merged**